src/components/libraryoptionseditor/libraryoptionseditor.js

```diff
@@ -347,41 +356,37 @@ define(['globalize', 'dom', 'emby-checkbox', 'emby-select', 'emby-input'], funct
         parent.querySelector('.imageFetchers').addEventListener('click', onImageFetchersContainerClick);
     }
 
-    function embed(parent, contentType, libraryOptions) {
+    export function embed(parent, contentType, libraryOptions) {
         currentLibraryOptions = {
             TypeOptions: []
         };
         currentAvailableOptions = null;
-        var isNewLibrary = null === libraryOptions;
+        const isNewLibrary = null === libraryOptions;
         isNewLibrary && parent.classList.add('newlibrary');
-        return new Promise(function(resolve, reject) {
-            var xhr = new XMLHttpRequest;
-            xhr.open('GET', 'components/libraryoptionseditor/libraryoptionseditor.template.html', true);
-            xhr.onload = function(e) {
-                var template = this.response;
+        return import('text!./libraryoptionseditor.template.html').then(({default: template}) => {
```

Comment:
Prefer replacing XMLHttpRequest with the Fetch API instead of this to prevent future refactoring.